dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoRyaInstanceDetailsRepository.java

```diff
@@ -0,0 +1,120 @@
+                    "the instance name that this repository is connected to. Make sure you're connected to the" +
+                    "correct Rya instance.");
+        }
+
+        if(isInitialized()) {
+            throw new AlreadyInitializedException("The repository has already been initialized for the Rya instance named '" +
+                    instanceName + "'.");
+        }
+
+        // Create the document that hosts the details if it has not been created yet.
+        final DBCollection col = db.createCollection(INSTANCE_DETAILS_COLLECTION_NAME, new BasicDBObject());
+
+        // Write the details to the collection.
+        col.insert(MongoDetailsAdapter.serialize(details));
+    }
+
+    @Override
+    public RyaDetails getRyaInstanceDetails() throws NotInitializedException, RyaDetailsRepositoryException {
+        // Preconditions.
+        if(!isInitialized()) {
+            throw new NotInitializedException("Could not fetch the details for the Rya instanced named '" +
+                    instanceName + "' because it has not been initialized yet.");
+        }
+
+        // Fetch the value from the collection.
+        final DBCollection col = db.getCollection(INSTANCE_DETAILS_COLLECTION_NAME);
+        //There should only be one document in the collection.
+        final DBObject mongoObj = col.findOne();
+        // Deserialize it.
+        return MongoDetailsAdapter.deserialize( mongoObj );
+    }
+
+    @Override
+    public void update(final RyaDetails oldDetails, final RyaDetails newDetails)
+            throws NotInitializedException, ConcurrentUpdateException, RyaDetailsRepositoryException {
+        // Preconditions.
+        requireNonNull(oldDetails);
+        requireNonNull(newDetails);
+
+        if(!newDetails.getRyaInstanceName().equals( instanceName )) {
+            throw new RyaDetailsRepositoryException("The instance name that was in the provided 'newDetails' does not match " +
+                    "the instance name that this repository is connected to. Make sure you're connected to the" +
+                    "correct Rya instance.");
+        }
+
+        if(!isInitialized()) {
+            throw new NotInitializedException("Could not update the details for the Rya instanced named '" +
+                    instanceName + "' because it has not been initialized yet.");
+        }
+
+        final DBCollection col = db.getCollection(INSTANCE_DETAILS_COLLECTION_NAME);
+        final DBObject oldObj = MongoDetailsAdapter.serialize(oldDetails);
+        final DBCursor cursor = col.find(oldObj);
+        if(cursor.count() == 0) {
+            throw new ConcurrentUpdateException("Could not update the details for the Rya instance named '" +
+                    instanceName + "' because the old value is out of date.");
+        }
+        final DBObject newObj = MongoDetailsAdapter.serialize(newDetails);
+        final WriteResult result = col.update(oldObj, newObj);
```

Comment:
If line 113 is a check and set which can fail if the old object could not be found because it is out of date, then you don't need lines 107 - 111. I thought you would throw a ConcurrentUpdateException if the result of the set and check returns that it didn't update anything.